src/extent.c

```diff
@@ -1024,6 +1119,12 @@ extent_dalloc_wrapper(tsdn_t *tsdn, pac_t *pac, ehooks_t *ehooks,
 			return;
 		}
 		extent_reregister(tsdn, pac, edata);
+		/* If failed to dalloc, restore the guard pages */
+		if (edata_guarded_get(edata)) {
+			san_guard_pages(tsdn, ehooks, edata, pac->emap,
+			    /* left */ true, /* right */ true,
```

Comment:
I'm not a big fan of having many bool args, but I don't know if creating helper functions for all used combinations (like `san_guard_pages_right`, etc.) is better.